src/main/java/com/ace/alfox/controllers/ActionController.java

```diff
@@ -3,32 +3,42 @@
 import com.ace.alfox.game.models.Player;
 import com.ace.alfox.lib.ActionFactory;
 import com.ace.alfox.lib.ActionResult;
-import com.ace.alfox.lib.PlayerFactory;
-import org.springframework.http.*;
+import com.ace.alfox.lib.Database;
+import org.dizitart.no2.NitriteId;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
 
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpSession;
-
-import java.io.*;
 import java.util.Map;
 
 @RestController
 class ActionController {
 
+    @Autowired
+    private Database db;
+    @Autowired
+    private ActionFactory actionFactory;
+
     @RequestMapping(value = "/perform/{command}", method = { RequestMethod.GET, RequestMethod.POST })
-    public ResponseEntity home(HttpServletRequest request, @PathVariable String command, @RequestBody Map<String, Object> params) throws IOException {
+    public ResponseEntity home(HttpServletRequest request, @PathVariable String command, @RequestBody Map<String, Object> params) {
         HttpSession session = request.getSession(true);
-        String playerId = (String) session.getAttribute("cid");
-
-        Player player = PlayerFactory.fetch(playerId);
-        if(player == null) {
-            return ResponseEntity.notFound().build();
+        NitriteId playerId = (NitriteId) session.getAttribute("pid");
+
+        if(playerId == null) {
+            //return ResponseEntity.notFound().build();
+            // Instead of just returning 404, create a test player
+            Player _player = new Player();
```

Comment:
What does the underscore represent? Is it meant to denote local variables or something?

Reply:
a habit picked up from javascript, yes a temporary variable only important to this block. In Java I should actually spell it out as such.